pages/users/signup.tsx

```diff
@@ -1,33 +1,180 @@
-// import { csrfToken, signIn } from 'next-auth/client';
-// import { NextPageContext, GetStaticProps } from 'next';
+import { useState } from 'react';
+import { Grid, TextField, Button } from '@material-ui/core';
+import { Formik, Form } from 'formik';
+import SignupSchema from 'interfaces/signup';
+import Layout from 'components/Layout';
+import { signIn } from 'next-auth/client';
+import styles from '../../styles/users/Signup.module.css';
+
+interface FormValues {
+  email: string;
+  password: string;
+  confirmPassword: string;
+}
+
+interface ErrorValues {
+  email?: string;
+  password?: string;
+  confirmPassword?: string;
+}
 
 // UserSignUp page. Will need additional email verification to be able to create organizations.
 const UserSignUp: React.FC = () => {
-  const handleSubmit = (e: any): void => {
-    console.log(e.target.value);
+  const [emailExists, setEmailExists] = useState(false);
+
+  const handleSubmit = async (values: FormValues): Promise<any> => {
+    // Make post request
+    const res: any = await fetch('/api/users/', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        email: values.email,
+        password: values.password,
+      }),
+    });
+
+    // Check if email exists
+    if (!res) {
+      setEmailExists(true);
+      return new Error('Email already exists.');
+    }
+
+    // Sign in user
+    try {
+      signIn('credentials', {
+        email: values.email,
+        password: values.password,
+        callbackUrl: 'http://localhost:3000',
+      });
+    } catch {
+      throw new Error('Could not sign in.');
+    }
+    return res;
   };
 
+  // Validates inputs
+  const validate = (values: FormValues): ErrorValues => {
+    const errors: { [k: string]: string } = {};
+    const { error } = SignupSchema.validate(values, {
+      context: { strict: true },
+      abortEarly: false,
+    });
+
+    // Fill out the errors payload with the correct messaging
+    if (error) {
+      for (let i = 0; i < error.details.length; i += 1) {
+        const val = error.details[i];
+        errors[val.path[0]] = val.message;
+      }
+    }
+    return errors;
+  };
+
+  const constructRow = (
+    title: string,
+    varName: string,
+    handleChange: any,
```

Comment:
handleChange type should be `FormikHandlers['handleChange']` imported from Formik